src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -114,8 +146,15 @@ public override void OnStopActivity(Activity activity, object payload)
                 var response = context.Response;
 
                 span.PutHttpStatusCode(response.StatusCode, response.HttpContext.Features.Get<IHttpResponseFeature>().ReasonPhrase);
+
+                // activity here will be the activity which we created using ActivitySource in the Start callback.
```

Comment:
> // activity here will be the activity which we created using ActivitySource in the Start callback. [](start = 15, length = 99)

how this magic is done? doesn't OnStopActivity is teh call back called from DiagnosticListener?

Reply:
diagnosticlistener callback itself doesn't give us activity. Instead we access it using Activity.Current.
https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/src/OpenTelemetry/Instrumentation/DiagnosticSourceListener.cs#L40

Since we started a new activity() in the OnBegin, Activity.Current will now be the one we created in OnBegin. This will have its .Parent set to the one created by "old" instrumentation code using DiagnosticSource.StartActivity

Reply:
make sense. thanks for the clarification. I am not expecting DiagnsoticListener will notify on the stop event of the Activity created from ActivitySource, right?

Reply:
Correct. (In fact diagnosticlistener will only notify for diagnosticsource.write events, never for activity.start or stop. On listening side, we get Activity.Current to get the activity)